scan-payment/src/main/java/com/getbouncer/scan/payment/Image.kt

```diff
@@ -86,159 +94,182 @@ fun hasOpenGl31(context: Context): Boolean {
 /**
  * Fragments the image into multiple segments and places them in new segments.
  */
-fun Bitmap.fragment(
-    fromSegments: Array<Rect>,
-    toSegments: Array<Rect>,
-    toSize: Size
+fun Bitmap.rearrangeBySegments(
+    fromSegments: Map<Rect, Rect>,
+    futureImageSize: Size
 ): Bitmap {
-    require(fromSegments.size == toSegments.size) {
-        "Number of source segments does not match number of destination segments"
-    }
-
     val current = this
-    val result = Bitmap.createBitmap(toSize.width, toSize.height, this.config)
+    val result = Bitmap.createBitmap(futureImageSize.width, futureImageSize.height, this.config)
     val canvas = Canvas(result)
 
-    (0 until fromSegments.size).map {
-        val image = current.crop(fromSegments[it]).scale(toSegments[it].size())
+    for ((from, to) in fromSegments) {
+        val image = current.crop(from).scale(to.size())
         canvas.drawBitmap(
             image,
-            toSegments[it].left.toFloat(),
-            toSegments[it].top.toFloat(),
+            to.left.toFloat(),
+            to.top.toFloat(),
             null
         )
     }
 
     return result
 }
 
+fun Bitmap.resizeRegion(
+    originalCenterRect: Rect,
+    futureCenterRect: Rect,
+    futureImageSize: Size
```

Comment:
nit: rename these parameters
```suggestion
    toCenterRect: Rect,
    toImageSize: Size
```